src/gui/qgsmaptoolcapture.cpp

```diff
@@ -1401,19 +1403,19 @@ void QgsMapToolCapture::cadCanvasReleaseEvent( QgsMapMouseEvent *e )
 
           if ( hasCurvedSegments && providerSupportsCurvedSegments )
           {
-            curveToAdd = captureCurve()->clone();
+            curveToAdd.reset( captureCurve()->clone() );
           }
           else
           {
-            curveToAdd = captureCurve()->curveToLine();
+            curveToAdd.reset( captureCurve()->curveToLine() );
           }
         }
         else
         {
-          curveToAdd = captureCurve()->clone();
+          curveToAdd.reset( captureCurve()->clone() );
         }
         QgsCurvePolygon *poly = new QgsCurvePolygon();
-        poly->setExteriorRing( curveToAdd );
+        poly->setExteriorRing( curveToAdd.get() );
```

Comment:
setExteriorRing takes ownership of curvetoadd so you have to release the unique_ptr.